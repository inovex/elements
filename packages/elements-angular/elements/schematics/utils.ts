import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
