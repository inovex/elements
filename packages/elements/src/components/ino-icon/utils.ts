import { getAssetPath } from '@stencil/core';

let CACHED_MAP: Map<string, string>;

export function getIconMap(): Map<string, string> {
	if (!CACHED_MAP) {
		const win = window as any;
		win.inoIcons = win.inoIcons || {};
		CACHED_MAP = win.inoIcons.map = win.inoIcons.map || new Map();
	}
	return CACHED_MAP;
}

export function addIcons(icons: { [name: string]: string }) {
	const map = getIconMap();
	Object.keys(icons).forEach((name) => map.set(name, icons[name]));
}

export const getSrc = (src: string | undefined) => {
	if (isStr(src)) {
		src = src.trim();
		if (isSrc(src)) {
			return src;
		}
	}
	return null;
};

export function getName(name: string | undefined) {
	if (name && !isSrc(name)) {
		return name;
	}
	return undefined;
}

export function getUrl(src: string | undefined, icon: string | undefined) {
	let url = getSrc(src);

	if (url) {
		return url;
	}
	url = getName(icon);
	if (url) {
		return getNamedUrl(url);
	}

	url = getSrc(icon);
	if (url) {
		return url;
	}

	return null;
}

export function getNamedUrl(name: string) {
	const url = getIconMap().get(name);
	if (url) {
		return url;
	}
	return getAssetPath(`./ino-icon/${name}.svg`);
}

export const isSrc = (str: string) => str.length > 0 && /(\/|\.)/.test(str);

export const isStr = (val: any): val is string => typeof val === 'string';

export const toLower = (val: string) => val.toLowerCase();
