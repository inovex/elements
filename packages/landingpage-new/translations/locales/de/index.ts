import learn from './learn';
import home from './home';
import about from './about';

export default {
  learn,
  home,
  about,
};
