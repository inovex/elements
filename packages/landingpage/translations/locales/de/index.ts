import about from './about.json';
import common from './common.json';
import explore from './explore.json';
import home from './home.json';
import library from './library.json';

export default {
  home,
  about,
  common,
  library,
  explore,
};
