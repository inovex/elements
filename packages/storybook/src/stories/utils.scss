/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
