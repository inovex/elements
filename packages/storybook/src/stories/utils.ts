import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
