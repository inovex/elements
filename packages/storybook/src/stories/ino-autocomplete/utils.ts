import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
