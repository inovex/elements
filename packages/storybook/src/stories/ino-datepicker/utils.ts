import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
