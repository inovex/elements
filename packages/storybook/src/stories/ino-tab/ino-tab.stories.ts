import { html } from 'lit-html';
import { defaultDecorator, withIconControl } from '../utils';
import { Story } from '@storybook/web-components';
import { Components } from '@inovex.de/elements';

export default {
  title: `Structure/ino-tab`,
  component: 'ino-tab',
  decorators: [defaultDecorator],
};

export const Playground: Story<Components.InoTab> = args => html`
  <ino-tab
    icon="${args.icon}"
    indicator-content-width="${args.indicatorContentWidth}"
    label="${args.label}"
    stacked="${args.stacked}">
  </ino-tab>
`;
Playground.args = {
  indicatorContentWidth: false,
  label: 'Customizable Tab',
  stacked: false,
}
withIconControl(Playground, 'icon', 'info');

export const Variants = () => html`
  <div class="story-tab-variants">
    <ino-tab label="Default" icon="info"></ino-tab>
    <ino-tab label="Stacked" icon="info" stacked></ino-tab>
    <ino-tab-bar active-tab="0">
      <ino-tab label="Default active Tab" icon="info"></ino-tab>
    </ino-tab-bar>
    <ino-tab-bar active-tab="0">
      <ino-tab
        label="Active content width indicator"
        icon="info"
        indicator-content-width
      >
      </ino-tab>
    </ino-tab-bar>
  </div>
`;
