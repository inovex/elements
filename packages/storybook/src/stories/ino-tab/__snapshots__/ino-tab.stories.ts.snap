// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-tab IndicatorContentWidth smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-tab icon="info"
             indicator-content-width="true"
             stacked="false"
             class="hydrated"
    >
      <button class="mdc-tab"
              role="tab"
              aria-selected="false"
      >
        <span class="mdc-tab__content">
          <ino-icon class="mdc-tab__icon ino-icon--color-scheme-primary hydrated">
          </ino-icon>
          <span class="mdc-tab__text-label">
            Label
          </span>
          <span class="mdc-tab-indicator">
            <span class="mdc-tab-indicator__content mdc-tab-indicator__content--underline">
            </span>
          </span>
        </span>
        <span class="mdc-tab__ripple">
        </span>
      </button>
    </ino-tab>
  </div>
</div>
`;

exports[`Structure/ino-tab Playground smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-tab icon="info"
             indicator-content-width="false"
             stacked="false"
             class="hydrated"
    >
      <button class="mdc-tab"
              role="tab"
              aria-selected="false"
      >
        <span class="mdc-tab__content">
          <ino-icon class="mdc-tab__icon hydrated ino-icon--color-scheme-primary">
          </ino-icon>
          <span class="mdc-tab__text-label">
            Label
          </span>
        </span>
        <span class="mdc-tab-indicator">
          <span class="mdc-tab-indicator__content mdc-tab-indicator__content--underline">
          </span>
        </span>
        <span class="mdc-tab__ripple">
        </span>
      </button>
    </ino-tab>
  </div>
</div>
`;

exports[`Structure/ino-tab Stacked smoke-test 1`] = `
<div id="root-inner">
  <div class>
    <ino-tab icon="info"
             indicator-content-width="false"
             stacked="true"
             class="hydrated"
    >
      <button class="mdc-tab mdc-tab--stacked"
              role="tab"
              aria-selected="false"
      >
        <span class="mdc-tab__content">
          <ino-icon class="mdc-tab__icon ino-icon--color-scheme-primary hydrated">
          </ino-icon>
          <span class="mdc-tab__text-label">
            Label
          </span>
        </span>
        <span class="mdc-tab-indicator">
          <span class="mdc-tab-indicator__content mdc-tab-indicator__content--underline">
          </span>
        </span>
        <span class="mdc-tab__ripple">
        </span>
      </button>
    </ino-tab>
  </div>
</div>
`;
